src/modules/tlsauth/README.md

```diff
@@ -0,0 +1,89 @@
+# TLSAuth module
+
+This modules integrates TLS client side certificates with the ACL subsystem,
+making it possible to automatically grant users their ACL permissions based on
+the client-side certificate used to authenticate.
+
+## Getting started
+
+To get started quickly and see how it works, follow these steps:
+
+1. Run `./utils/gen-test-certs.sh` to have server certs generated.
+
+2. Create user certificates for Alice and Bob. We will use the commonName (CN)
+   attribute to store the user identity, and a "Redis Test" organization and
+   "Users" organizatinalUnit.
+
+        openssl req \
+                 -new -sha256 \
+                 -newkey rsa:2048 -nodes -keyout tests/tls/alice.key \
+                 -subj '/O=Redis Test/OU=Users/CN=alice' | \
+                     openssl x509 \
+                         -req -sha256 -CA tests/tls/ca.crt \
+                         -CAkey tests/tls/ca.key \
+                         -CAserial tests/tls/ca.txt \
+                         -CAcreateserial \
+                         -days 365 \
+                         -out tests/tls/alice.crt
+
+        openssl req \
+                 -new -sha256 \
+                 -newkey rsa:2048 -nodes -keyout tests/tls/bob.key \
+                 -subj '/O=Redis Test/OU=Users/CN=bob' | \
+                     openssl x509 \
+                         -req -sha256 -CA tests/tls/ca.crt \
+                         -CAkey tests/tls/ca.key \
+                         -CAserial tests/tls/ca.txt \
+                         -CAcreateserial \
+                         -days 365 \
+                         -out tests/tls/bob.crt
+
+3. Create a Redis configuration with two ACL users and TLS enabled:
+
+         tls-ca-cert-file "tests/tls/ca.crt"
+         tls-cert-file "tests/tls/redis.crt"
+         tls-key-file "tests/tls/redis.key"
+         tls-auth-clients yes
+         tls-port 6379
+         port 0
+         user alice on #93dbc7782132ec0d34e8543805a5f85401044c0e04b6c69d4544cac7d29eb029 ~* +@all
+         user bob on #2a31d2ffaa7fe4e35b7fe6c990c02489ebee5a18cf388c3236910c59212e11b3 ~* +@all
+         loadmodule src/modules/tlsauth/tlsauth.so attr-user=commonName attr-check=O:"Redis Test" attr-check=OU:Users
```

Comment:
It's a bit odd how much of a juxtaposition this is with normal redis commands/arguments. The arguments have complex delimiters and are case sensitive. I would expect something like:

```
loadmodule src/modules/tlsauth/tlsauth.so USER-ATTRIBUTE commonName CHECK-ATTRIBUTE organization "Redis Test" CHECK-ATTRIBUTE OU Users 
```

Just a thought if we are setting a standard for "official" modules.

Reply:
Totally agree, it's much more readable (and even easier to parse) this way.